sys/riscv/include/atomic.h

```diff
@@ -104,10 +104,25 @@ atomic_fcmpset##SUFFIX##WIDTH(__volatile uint##WIDTH##_t *p,		\
 	ATOMIC_FCMPSET_ORDER(WIDTH, _, __ATOMIC_RELAXED)		\
 	ATOMIC_FCMPSET_ACQ_REL(WIDTH)					\
 
+#ifdef __CHERI_PURE_CAPABILITY__
+/*
+ * The purecap kernel can not use the generic sub-word atomics
+ * as they break in the presence of subobject bounds.
+ */
+ATOMIC_CMPSET(8);
+ATOMIC_FCMPSET(8);
+ATOMIC_CMPSET(16);
+ATOMIC_FCMPSET(16);
+#define	atomic_cmpset_8 atomic_cmpset_8
+#define	atomic_fcmpset_8 atomic_fcmpset_8
+#define	atomic_cmpset_16 atomic_cmpset_16
+#define	atomic_fcmpset_16 atomic_fcmpset_16
```

Comment:
```suggestion
#define	atomic_cmpset_8		atomic_cmpset_8
#define	atomic_fcmpset_8		atomic_fcmpset_8
#define	atomic_cmpset_16		atomic_cmpset_16
#define	atomic_fcmpset_16	atomic_fcmpset_16
```

(Adjust the tabs as needed to line up the column with `atomic_cmpset_char` below)